### ISSUE TITLE
Add accessibility settings for syntax highlighting in preferences

### ISSUE THREAD

**Issue by vkbo** [vkbo/novelWriter#2691]
> About your suggestion for other ways of indicating difference: There is already an option to just show emphasis rather than or in addition to use colour. This can be applied to more features I think. It is trivial to fix it in the syntax highlighter as it just applies pre-defined formats.
> 
> How about we add an Accessibility section below the theme section, and add some more options like that. For instance, replace tags and shortcodes with **bold**. I can add a dropdown of options with, say bold or underline with and without colour and just plain colour as the default. I can move the colouring of emphasis to the same section. I can also add a bold flag for synopsis/note/story comments with and without colour.
> 
> With this, we can reserve the clear colour differences for dialogue and notes.
> 
> Does that sound like a solution? 

 _Originally posted by @vkbo in [#2689](https://github.com/vkbo/novelWriter/issues/2689#issuecomment-4108342790)_

**Comment by vkbo** [comment]
I'm a little uncertain how helpful these settings are. I'm considering instead adding a simple setting that adds a dotted line under all formatting tags and modifiers. It seems to me to be the most visually helpful option. Bold sort of just drowns in the text.

Here's an example with both bold and dotted for comments, shortcodes and replace tags on a colourless theme:

<img width="839" height="793" alt="Image" src="https://github.com/user-attachments/assets/87349e95-4e1a-4b9b-bf5e-4b27c01eaf63" />

What do you think, @paulkater?

**Comment by vkbo** [comment]
Here's the basic colour plus dotted applied to the default theme. I will make new colour themes anyway, so the colour is less relevant here. The colourblindness themes will have fewer colours anyway.

<img width="839" height="793" alt="Image" src="https://github.com/user-attachments/assets/aa1d8917-48f7-4b99-837c-facafdef4e04" />

**Comment by paulkater** [comment]
I really appreciate your efforts in this, and I love the dotted line bit.
I assume here that this is quite a contrast rich theme for you, or am I wrong here?  I ask this because for me this isn't contrast rich. <grimace> For me the examples up here look like lots of shades of grey.

I wonder how this would look in a white background (like here on github) and black letters (like here on github) and the colour for dialogue in clue (like the commit button here on github).

e.g.
<img width="270" height="155" alt="Image" src="https://github.com/user-attachments/assets/6a60bf59-77bb-424f-ad1c-302d1e1134d6" />

or this, as I use in novelWriter now:

<img width="446" height="140" alt="Image" src="https://github.com/user-attachments/assets/0bfe8ba1-475e-40f7-9edd-43809239819e" />

It's really that simple and the reason I'm still on this older version.

This:

<img width="314" height="72" alt="Image" src="https://github.com/user-attachments/assets/a4b46c94-4cde-428f-8023-6f5bb4f86538" />
does show a different colour in the quotes but it kind of 'fizzles away' against the grey background and loses the 'punch' among the light grey lettering.

Gosh, I had no idea this would be so difficult to explain, I'm very sorry.

**Comment by vkbo** [comment]
This issue is about the dotted line, not the colour. The first screenshot I posted is not coloured at all. The second uses a muted coloured theme called "Tomorrow". But I did not intend to use that for the colour theme. I'm making a new theme based on the old default theme, and reducing the colours used two a couple of shades of blue and yellow, which should work for both red and green colour blindness.

In any case, I will merge the feature with the dotted line then. It seems useful for other use cases as well.
